### DISCUSSION TITLE
Do you plan to add layers over your map?

### DISCUSSION THREAD

**Discussion by jesussmile** [liodali/osm_flutter#391] [General]
Hi Liodali,
do you plan to add layers over your map? i can see that you are rendering everything over the native view. one benifit is it will be fast,  but other openstreetmap plugin dont do that , they will just add a layer such that it is easier to add marker , make it on tap, then you can add image layer also on top too

I have been working with `flutter_map` below  is a screen shot of adding polygon and images etc..it works fine everything has a onTap listener but it is slow. your map is very fast. other plugin can't handle multiple markers may be over 1000 , may be you could just add a layer and code markers on top with flutter ? can you give me a reason why you chose not to do it this way ?

![IMG_0034](https://user-images.githubusercontent.com/11044978/226221731-336f8fbb-1ca7-400f-aacf-e13543dc7a17.PNG)

**Comment by liodali** [comment]
we already use layers but natively

**Reply by liodali** [reply]
create marker in flutter and let native view controlled little bit complex because we can have x,y different of what we have in flutter view and also our SDKs dont support custom view for marker
for example in android or ios we can only put image as marker
